### DISCUSSION TITLE
What is the purpose of the `inspect` function in DriverAdapter?

### DISCUSSION THREAD

**Discussion by yongil-XCENA** [facebookincubator/velox#10455] [Q&A]
Hello.
My company MetisX is developing a HW accelerator for OLAP and we are trying to accelerate Velox with our HW.
Referring to the Wave code, I realized that we can implement it by utilizing DriverAdapter.
But I don't quite understand the purpose of the `inspect` function in DriverAdapter. Is there any documentation or example code somewhere that explains this?
Actually, what I want to do is get a complete PlanFragment and have it run through our acceleration engine path. I know from the VeloxCon2024 video that NeuroBlade is taking this approach. I'm guessing I could use the `inspect` function of the DriverAdapter, is my approach correct?

**Comment by karthikeyann** [comment]
`inspect` can take const PlanFragment argument and it's up to the `inspect` to what to do with it. `inspect` can choose to cache the PlanFragment. Later this cached PlanFragment can be utilized by driverAdapter or driver. driver object will execute single pipeline. It can not execute the entire PlanFragment in one go by design. (but this can be overridden by driverAdapter by editing the pipeline operators).
